modules/deployment-model/src/test/java/org/mule/runtime/deployment/model/internal/plugin/SuccessfulAbstractPluginTestCase.java

```diff
@@ -0,0 +1,225 @@
+      throws MalformedPluginException, MalformedClassloaderModelException {
+    PluginDescriptor pluginDescriptor = getSuccessfulPlugin("plugin-dependencies-no-exportedPackages");
+
+    assertClassloaderDescriptorJustResources(pluginDescriptor);
+    assertNoExtensionModelDescriptor(pluginDescriptor);
+  }
+
+  @Test
+  public void testSuccessfulDeploymentModelDependenciesNoExportedPackagesAndNoExportedResources()
+      throws MalformedPluginException, MalformedClassloaderModelException {
+    PluginDescriptor pluginDescriptor =
+        getSuccessfulPlugin("plugin-dependencies-no-exportedPackages-and-no-exportedResources");
+
+    assertClassloaderDescriptorNoResourcesNoPackages(pluginDescriptor);
+    assertNoExtensionModelDescriptor(pluginDescriptor);
+  }
+
+  @Test
+  public void testSuccessfulDeploymentModelDependenciesNoExportedResources()
+      throws MalformedPluginException, MalformedClassloaderModelException {
+    PluginDescriptor pluginDescriptor = getSuccessfulPlugin("plugin-dependencies-no-exportedResources");
+
+    assertClassloaderDescriptorJustPackages(pluginDescriptor);
+    assertNoExtensionModelDescriptor(pluginDescriptor);
+  }
+
+  @Test
+  public void testSuccessfulDeploymentModelEmpty() throws MalformedPluginException, MalformedClassloaderModelException {
+    PluginDescriptor pluginDescriptor = getSuccessfulPlugin("plugin-empty");
+
+    assertClassloaderDescriptor(pluginDescriptor);
+    assertNoExtensionModelDescriptor(pluginDescriptor);
+  }
+
+  @Test
+  public void testSuccessfulDeploymentModelLibs() throws MalformedPluginException, MalformedClassloaderModelException {
+    String pluginFolder = "plugin-libs";
+    PluginDescriptor pluginDescriptor = getSuccessfulPlugin(pluginFolder);
+
+    assertClassloaderDescriptor(pluginDescriptor);
+    assertExtensionModelDescriptor(pluginDescriptor);
+  }
+
+  @Test
+  public void testSuccessfulDeploymentModelLibsAndClasses()
+      throws MalformedPluginException, MalformedClassloaderModelException {
+    String pluginFolder = "plugin-libs-and-classes";
+    PluginDescriptor pluginDescriptor = getSuccessfulPlugin(pluginFolder);
+
+    assertClassloaderDescriptor(pluginDescriptor);
+    assertExtensionModelDescriptor(pluginDescriptor);
+  }
+
+  private void assertMinimalRequirementsPlugin(PluginDescriptor pluginDescriptor, String pluginParentFolder) {
+    String expectedPluginName = getPluginName(pluginParentFolder);
+    Assert.assertThat(pluginDescriptor.getName(), Is.is(expectedPluginName));
+    Assert.assertThat(pluginDescriptor.getMinMuleVersion(), Is.is(new MuleVersion("4.0.0")));
+  }
+
+  protected String getPluginName(String pluginParentFolder) {
+    return hasJsonDescriptor ? "plugin-name" : pluginParentFolder;
+  }
+
+  protected void assertClassloaderDescriptor(PluginDescriptor pluginDescriptor) {
+    Descriptor classloaderModelDescriptor = pluginDescriptor.getClassloaderModelDescriptor();
+
+    if (hasJsonDescriptor) {
```

Comment:
You are trying to manage two different set of scenarios in only one class. Every assertX method has does different things depending on whether you have a "json descriptor". You should split the test class in two